src/window/glfwWindow.cpp

```diff
@@ -0,0 +1,140 @@
+// clang-format off
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+// clang-format on
+#include <cassert>
+#include <iostream>
+#include <unordered_map>
+#include "singleton.h"
+#include "glfwWindow.h"
+
+struct KeyCallbackStruct {
+private:
+  std::unordered_map<GLFWwindow *, GWindowMgr *> windowMap;
+
+public:
+  void insertNewWindow(GWindowMgr *window);
+  GWindowMgr *getWindowMgr(GLFWwindow *window);
+  static void keyCallback(GLFWwindow *window, int key, int scancode, int action,
+                          int mods);
+};
+
+typedef SingletonBase<KeyCallbackStruct> Singleton;
```

Comment:
Ended up requiring a singleton here b\c glfw is a c api and c doesn't allow you to pass lambdas as a function ptr, which means I couldn't do a lamdba capture of the KeyCallBacks vector.